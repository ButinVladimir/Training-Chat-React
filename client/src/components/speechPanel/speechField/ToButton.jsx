import React from 'react';
import PropTypes from 'prop-types';

function ToButton({ to, onRemoveTo }) {
  return (
    <button className="to-button" type="button" onClick={onRemoveTo}>
      {`@${to}`}
    </button>
  );
}

ToButton.propTypes = {
  to: PropTypes.string.isRequired,
  onRemoveTo: PropTypes.func.isRequired,
};

export default ToButton;
