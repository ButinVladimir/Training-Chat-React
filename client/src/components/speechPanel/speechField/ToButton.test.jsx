import React from 'react';
import { shallow } from 'enzyme';
import ToButton from './ToButton';

describe('ToButton', () => {
  it('renders without crashing', () => {
    const component = shallow(<ToButton to="recipient" onRemoveTo={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  it('handles click', () => {
    const mock = jest.fn();
    const component = shallow(<ToButton to="recipient" onRemoveTo={mock} />);
    component.find('button').simulate('click');

    expect(mock).toHaveBeenCalledTimes(1);
    expect(mock).toHaveBeenCalledWith('recipient');
  });
});
