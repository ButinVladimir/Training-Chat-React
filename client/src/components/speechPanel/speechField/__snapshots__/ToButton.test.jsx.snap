// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToButton renders without crashing 1`] = `
<button
  className="to-button"
  onClick={[Function]}
  type="button"
>
  @recipient
</button>
`;
